### PR TITLE
Feature/update api client

### PR DESCRIPTION
**Pull request by MarkCherepovskyi**

### CODE REVIEW

File: packages/jac/src/json-api.test.ts [hunk 1]
```diff
@@ -127,6 +135,28 @@ describe('performs JsonApiClient request unit test', () => {
...
+
+    expect(error instanceof FetcherError<JsonApiResponseErrors>).toEqual(false)
+  })
+
+  test('should return true if error is instance of FetcherError', async () => {
+    const error = new FetcherError<JsonApiResponseErrors>(
+      {} as FetcherResponse<JsonApiResponseErrors>,
+    )
+    expect(error instanceof FetcherError<JsonApiResponseErrors>).toEqual(true)
+  })
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts]: useless tests. same as `expect(true).toEqual(true)`

File: packages/jac/src/tests/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export * from './mock-wrapper'
+export { default as ERROR_RESPONSE } from './mocks/fetcher-error-response.json'
```
Comment (napalmpapalam) [inline packages/jac/src/tests/index.ts]: where it used ? this just shows that u didn't really check your changes before sending it for review

File: packages/jac/src/tests/mocks/fetcher-error-response.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "errors": [
+    {
+      "id": "1",
+      "code": "err_some_code",
+      "status": "401",
+      "title": "Unauthorized"
+    }
+  ]
+}
```
Comment (napalmpapalam) [inline packages/jac/src/tests/mocks/fetcher-error-response.json]: change.log, version bump https://github.com/distributed-lab/web-kit?tab=readme-ov-file#check-release-version

File: packages/jac/src/json-api.test.ts [hunk 4]
```diff
@@ -127,6 +135,28 @@ describe('performs JsonApiClient request unit test', () => {
     })
   })
 
+  test('should throw custom error', async () => {
+    const api = new JsonApiClient(VALID_CFG)
+
+    const error = new CustomError('')
+    jest.spyOn(api, 'request').mockRejectedValue(error)
+
+    await expect(api.get('')).rejects.toThrowError(CustomError)
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts]: a u sure that await is needed here ?

File: packages/jac/src/json-api.test.ts [hunk 5]
```diff
@@ -26,7 +34,7 @@ describe('performs JsonApiClient request unit test', () => {
     test('should throw exception "baseUrl" argument not provided', () => {
       const api = new JsonApiClient({ baseUrl: VALID_BASE_URL_1 })
       expect(() => api.useBaseUrl('')).toThrow(
-        'Fetcher: invalid base URL. TypeError [ERR_INVALID_URL]: Invalid URL',
+        'Fetcher: invalid base URL. TypeError: Invalid URL',
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts]: why did you change this?

File: packages/jac/src/json-api.test.ts [hunk 6]
```diff
@@ -16,6 +22,8 @@ const mockedBody = {
   },
 }
 
+export class CustomError extends RuntimeError {}
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts]: why it's exported ?

File: packages/jac/src/json-api.test.ts [hunk 7]
```diff
@@ -127,6 +135,28 @@ describe('performs JsonApiClient request unit test', () => {
     })
   })
 
+  test('should throw custom error', async () => {
+    const api = new JsonApiClient(VALID_CFG)
+
+    const error = new CustomError('')
+    jest.spyOn(api, 'request').mockRejectedValue(error)
+
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts]: this test actually does not cover your use case. maybe partly. mock response with error response (400 or some other (u could refer to the example below how to do it)) and use interceptor which will throw your custom error. 

```ts
 test('should return correct data', () => {
   const rawResponse = MockWrapper.makeFetcherResponse({
    errors: [{
       id: "1",
       code: "err_some_code",
       status: "401",
       title: "Unauthorized"
    }]
  }, 401)
   jest.spyOn(Fetcher.prototype, 'request').mockResolvedValue(rawResponse)
   const api = new JsonApiClient(VALID_CFG)
   return api.get('').then(({ data }) => expect(data).toEqual(PARSED_RESPONSE))
  })
```

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/jac/src/json-api.test.ts (hunk 1 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/json-api.test.ts]: removed

File: packages/jac/src/tests/index.ts (hunk 2 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/tests/index.ts]: removed

File: packages/jac/src/json-api.test.ts (hunk 7 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/json-api.test.ts]: updated

File: packages/jac/src/json-api.test.ts (hunk 6 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/json-api.test.ts]: fixed

File: packages/jac/src/json-api.test.ts (hunk 5 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/json-api.test.ts]: I had problems  with  this running tests on node 21.6.  Fixed by  using 18.19

File: packages/jac/src/json-api.test.ts (hunk 4 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/json-api.test.ts]: fixe

File: packages/jac/src/tests/mocks/fetcher-error-response.json (hunk 3 above)
Comment (MarkCherepovskyi) [reply packages/jac/src/tests/mocks/fetcher-error-response.json]: updated

Comment (napalmpapalam) [reply packages/jac/src/tests/mocks/fetcher-error-response.json]: u didn't bump all packages version

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [1.0.0-rc.10] - 2024-01-17
+### Fixed
+- `@distributedlab/jac` - possibility to throw custom error on request
+
+
```
Comment (napalmpapalam) [inline CHANGELOG.md]:
```suggestion
```

File: packages/jac/src/json-api.test.ts [hunk 9]
```diff
@@ -127,6 +137,70 @@ describe('performs JsonApiClient request unit test', () => {
...
+    )
+
+    const api = new JsonApiClient(VALID_CFG)
+
+    jest
+      .spyOn(Fetcher.prototype, 'request')
+      .mockRejectedValue(new FetcherError(rawErrorResponse))
+
+    expect(api.get('')).rejects.toThrowError(errors.UnauthorizedError)
+  })
```
Comment (napalmpapalam) [inline packages/jac/src/json-api.test.ts:181]: this cases have already covered there https://github.com/distributed-lab/web-kit/blob/main/packages/jac/src/middlewares/parse-json-api-error.test.ts

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

**Review by napalmpapalam — APPROVED** [review]
